### ISSUE TITLE
Failed onboarding with "am" module error

### ISSUE THREAD

**Issue by nahun** [F5Networks/f5-declarative-onboarding#91]
Using BIG-IQ to onboard a new BIG-IP gets an error.

DO Version: tested on 1.7.0 and 1.8.0 on BIG-IQ
BIG-IQ Version: 7.0.0-0.0.1854
BIG-IP Version: 14.1.2.2

Error:
```
Failed to complete onboarding task. code: 500, message: invalid config - rolled back, error: am is not provisionable. Provisionable modules are: afm
,apm,asm,avr,dos,fps,gtm,ilx,lc,ltm,pem,sslo,swg,urldb
```

Here is the declaration sanitized a bit:
```json
{
    "class": "DO",
    "declaration": {
        "schemaVersion": "1.8.0",
        "class": "Device",
        "async": true,
        "Common": {
            "class": "Tenant",
            "licenseregkey": {
                "class": "License",
                "licenseType": "regKey",
                "regKey": "<SNIP>",
                "overwrite": true
            },
            "provisioning": {
                "class": "Provision",
                "ltm": "nominal"
            },
            "configsyncip": {
                "class": "ConfigSync",
                "configsyncIp": "10.0.0.1"
            },
            "failoverAddress": {
                "class": "FailoverUnicast",
                "address": "10.0.0.1"
            },
            "Local_Group": {
                "class": "DeviceGroup",
                "type": "sync-failover",
                "members": ["bigip1", "bigip2"],
                "owner": "/Common/Local_Group/members/0",
                "autoSync": false,
                "networkFailover": true
            },
            "deviceTrust": {
                "class": "DeviceTrust",
                "localUsername": "admin",
                "localPassword": "admin",
                "remoteHost": "/Common/Local_Group/members/0",
                "remoteUsername": "admin",
                "remotePassword": "admin"
            },
            "configsync": {
                "class": "VLAN",
                "tag": "4094",
                "interfaces": [{"tagged": false, "name": "8.0"}]
            },
            "fw_transit": {
                "class": "VLAN",
                "tag": "2211",
                "interfaces": [{"tagged": true, "name": "5.0"}, {"tagged": true, "name": "6.0"}]
            },
            "isp1_transit": {
                "class": "VLAN",
                "tag": "2011",
                "interfaces": [{"tagged": true, "name": "5.0"}]
            },
            "isp2_transit": {
                "class": "VLAN",
                "tag": "2012",
                "interfaces": [{"tagged": true, "name": "6.0"}]
            },
            "configsync_ltm01": {
                "class": "SelfIp",
                "address": "10.0.0.1/24",
                "vlan": "configsync", 
                "allowService": "default",
                "trafficGroup": "traffic-group-local-only"
            },
            "fw_transit_ltm01": {
                "class": "SelfIp",
                "address": "172.16.0.2/27",
                "vlan": "fw_transit", 
                "allowService": "none",
                "trafficGroup": "traffic-group-local-only"
            },
            "fw_transit": {
                "class": "SelfIp",
                "address": "172.16.0.1/27",
                "vlan": "fw_transit", 
                "allowService": "none",
                "trafficGroup": "traffic-group-1"
            },
            "isp1_transit_ltm01": {
                "class": "SelfIp",
                "address": "34.2.2.2/28",
                "vlan": "isp1_transit", 
                "allowService": "none",
                "trafficGroup": "traffic-group-local-only"
            },
            "isp1_transit": {
                "class": "SelfIp",
                "address": "34.2.2.1/28",
                "vlan": "isp1_transit", 
                "allowService": "none",
                "trafficGroup": "traffic-group-1"
            },
            "hostname": "bigip1",
            "snmpAgent": {
                "class": "SnmpAgent",
                "allowList": ["127.0.0.0/8"]
            }
        }
    },
    "targetUsername": "admin",
    "targetHost": "172.16.1.10",
    "targetPassphrase": "admin",
    "bigIqSettings": {
        "clusterName": "Local_Group",
        "failImportOnConflict": false,
        "conflictPolicy": "USE_BIGIQ",
        "deviceConflictPolicy": "USE_BIGIP",
        "versionedConflictPolicy": "KEEP_VERSION",
        "statsConfig": {
            "enabled": false
        }
    }
}

```

**Comment by nahun** [comment]
This is happening on BIG-IP hardware i4600s and i4800s. It doesn't happen on VEs in VMware it seems.

**Comment by 464d41** [comment]
+1 . declaration fails for i5800 with similar error message if provisioning related declaration piece exists:
```
...
"myProvisioning": {
                "class": "Provision",
                "asm": "nominal"
            },
...
```

Error:
```
"errors": [
            "am is not provisionable. Provisionable modules are: afm,apm,asm,avr,cgnat,dos,fps,gtm,ilx,lc,ltm,pem,sslo,swg,tam,urldb,vcmp"
        ]
```

Declaration succeeds if provisioning is not declared.

**Comment by dstokesf5** [comment]
Added to our internal product backlog as AUTOTOOL-1094.

**Comment by stanislaspiron** [comment]
Same issue on vCMP guest on Viprion... but with "pem" module missing

`        "myProvisioning": {
            "class": "Provision",
            "ltm": "nominal"
            },`

Error:

` "result": {
        "class": "Result",
        "code": 500,
        "status": "ERROR",
        "message": "invalid config - rolled back",
        "errors": [
            "pem is not provisionable. Provisionable modules are: afm,am,apm,asm,avr,dos,fps,gtm,ilx,lc,ltm,swg,urldb",
            "pem is not provisionable. Provisionable modules are: afm,am,apm,asm,avr,dos,fps,gtm,ilx,lc,ltm,swg,urldb"
        ]
    },`

TMOS version 13.1.1.5
DO version : 1.9 , 1.10 (installed on the vcmp guest)
Hardware : Viprion C2400 / B2100

**Comment by LeonardosGitHub** [comment]
I'm having the same issue on vCMP guest on i10800:

        "myProvisioning": { "class": "Provision", "ltm": "nominal", "gtm": "nominal"}
Error:
"result": { "class": "Result", "code": 202, "status": "ROLLING_BACK", "message": "invalid config - rolling back", "errors": ["pem is not provisionable. Provisionable modules are: afm,am,apm,asm,avr,cgnat,dos,fps,gtm,ilx,lc,ltm,sslo,swg,urldb"]}

TMOS version: 15.1.0.1
DO version: 1.10
Hardware: i10800 vCMP guest

**Comment by dstokesf5** [comment]
This issue has been resolved as part of the 1.12.0 release.
